crates/sui-graphql-rpc/src/types/error.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::{Error, ErrorExtensions};
+
+#[derive(Debug, thiserror::Error)]
+pub enum CustomError {
+    #[error("{0}")]
+    Input(String),
+    #[error("{0}")]
+    ClientFetch(String),
+}
+
+impl ErrorExtensions for CustomError {
+    fn extend(&self) -> Error {
+        Error::new(format!("{}", self)).extend_with(|_err, e| match self {
+            CustomError::Input(_) => {
+                e.set("code", "INVALID_INPUT");
+            }
+            CustomError::ClientFetch(_) => {
+                e.set("code", "CLIENT_FETCH_ERROR");
+            }
```

Comment:
In #13378, I create a place to define these, let's add these codes there.  Also, let's use the [standard set](https://www.apollographql.com/docs/apollo-server/data/errors/#built-in-error-codes) of codes as much as possible, (e.g. `BAD_USER_INPUT` instead of `INVALID_INPUT`, `INTERNAL_SERVER_ERROR` instead of `CLIENT_FETCH_ERROR`).